File: C programs/pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

// Readability program //
// cs50 2020; problem set 2 ; problem 1 //

int main(void)
{
    string text = get_string("text\n");
    
    int words = 1, sentences = 0, letters = 0;
    
    float index = 0;
    
// finding word, sentences and letters quantities //

    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    
        else if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            sentences++;
        }   
    
        else if (isupper(text[i]) || islower(text[i]))
        {
            letters++;
        }
    }
    
    
    // classifing process (coleman - leau index //
    
    index = 0.0588 * (100 * (float) letters / (float) words) - 0.296 * (100 * (float) sentences / (float)words) - 15.8;
    
    if (index <= 0)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    
    else
    {
        printf("Grade %.0f\n", round(index));   
    }
    
}

File: Python Programs/pset6/readability/readability.py
# Prompts the user for the text and get the text length
text = input("Text: ")
text_len = len(text)

# Variables used to count words, sentences and letters
# Words start at 1 because the for loop does not count the last word since there's no space after it
words = 1
sentences = 0
setence_list = [".", "?", "!"]
letters = 0
index = 0

# Loop through the text collecting quantity of words, sentences and letters
for i in range(text_len):
    if (text[i] == " "):
        words += 1
    if (text[i] in setence_list):
        sentences += 1
    if (str.isalpha(text[i])):
        letters += 1

# implement Coleman-Liau index 
index = 0.0588 * (100 * letters / words) - 0.296 * (100 * sentences / words) - 15.8
index = round(index)

# Checks for the user's grade
if (index >= 16):
    print("Grade 16+")
elif (index < 1):
    print("Before Grade 1")
else:
    print(f"Grade {index}")